From d4f7a49d77bd8f4ac871a999fc9ec898cb22b8c3 Mon Sep 17 00:00:00 2001
From: Vlad Ivanov <vlad-mbx@ya.ru>
Date: Tue, 9 Feb 2016 10:35:23 +0300
Subject: [PATCH] windows_usb: enable RAW_IO policy by default

libusb should set RAW_IO policy unconditionally because it
implements the buffer management itself.

Signed-off-by: Vlad Ivanov <vlad-mbx@ya.ru>
---
 libusb/os/windows_usb.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libusb/os/windows_usb.c b/libusb/os/windows_usb.c
index 6640ad5..12fdab0 100644
--- a/libusb/os/windows_usb.c
+++ b/libusb/os/windows_usb.c
@@ -2890,6 +2890,10 @@ static int winusbx_configure_endpoints(int sub_api, struct libusb_device_handle
 			AUTO_CLEAR_STALL, sizeof(UCHAR), &policy)) {
 			usbi_dbg("failed to enable AUTO_CLEAR_STALL for endpoint %02X", endpoint_address);
 		}
+		if (!WinUSBX[sub_api].SetPipePolicy(winusb_handle, endpoint_address,
+			RAW_IO, sizeof(UCHAR), &policy)) {
+			usbi_dbg("failed to enable RAW_IO for endpoint %02X", endpoint_address);
+		}
 	}
 
 	return LIBUSB_SUCCESS;
-- 
2.5.0

